app/src/main/java/org/fossasia/pslab/fragment/HomeFragment.java

```diff
@@ -82,6 +89,7 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
 
         if (deviceConnected) {
+            progressDialog.show();
```

Comment:
This would create progress dialog when switching between fragments.
Try showing progress dialog when read version string from device.